File: hotel/hotel/HotelReservationCompleted.go
package hotel

import (
	"time"
)

type HotelReservationCompleted struct{
	EventType string	`json:"eventType" type:"string"`
	TimeStamp string 	`json:"timeStamp" type:"string"`
	Id int `json:"id" type:"int"` 
	ReservationId string `json:"reservationId" type:"string"` 
	UserId string `json:"userId" type:"string"` 
	StartDate  `json:"startDate" type:""` 
	EndDate  `json:"endDate" type:""` 
	Price int `json:"price" type:"int"` 
	Status string `json:"status" type:"string"` 
	RoomCount int `json:"roomCount" type:"int"` 
	HotelName string `json:"hotelName" type:"string"` 
	Address string `json:"address" type:"string"` 
	RoomNumber int `json:"roomNumber" type:"int"` 
	
}

func NewHotelReservationCompleted() *HotelReservationCompleted{
	event := &HotelReservationCompleted{EventType:"HotelReservationCompleted", TimeStamp:time.Now().String()}

	return event
}

File: reservation/src/main/java/product/domain/HotelReservationCompleted.java
package product.domain;

import java.time.LocalDate;
import java.util.*;
import lombok.*;
import product.domain.*;
import product.infra.AbstractEvent;

//<<< DDD / Domain Event
@Data
@ToString
public class HotelReservationCompleted extends AbstractEvent {

    private Long id;
    private String reservationId;
    private String userId;
    private Date startDate;
    private Date endDate;
    private Integer price;
    private String status;
    private Integer roomCount;
    private String hotelName;
    private String address;
    private Integer roomNumber;

    public HotelReservationCompleted(HotelReservation aggregate) {
        super(aggregate);
    }

    public HotelReservationCompleted() {
        super();
    }
}
//>>> DDD / Domain Event
